public/js/app.js

```diff
@@ -133,6 +141,15 @@ function parseGeoJSON(geojson){
       location['lng'] = coordinates[0];
       location['lat'] = coordinates[1];
 
+      var bookingLocationId = location['booking_location_id'];
+
+      if(bookingLocationId && bookingLocationId != '') {
+        var bookingLocation = findLocationProperties(features, bookingLocationId);
```

Comment:
Perhaps a bit of defensiveness in case an invalid ID creeps in? (I know there are tests to prevent against it, but still...)